### PR TITLE
Fine-tune 'Accept' button in AirWingConfigurationDiaglog

### PR DESCRIPTION
**Pull request by Raffson**
- Fixes 'Accept' not saving changes
- Adds confirmation pop-up when pressing [X] when a change was made, asking if you're sure you want to discard the changes and start the campaign
- Adds 'Reset' button reverting all changes made so far

### CODE REVIEW

**Comment by Raffson** [comment]
Fix #2325

**Comment by SnappyComebacks** [comment]
Was there a reason the window close (the X button) has a dialogue "Are you sure you want to discard" but the big red discard button does not?

Otherwise it looks like things are working.

**Comment by Raffson** [comment]
> Was there a reason the window close (the X button) has a dialogue "Are you sure you want to discard" but the big red discard button does not?
> 
> 
> 
> Otherwise it looks like things are working.

Yes there was, me and @Starfire13 were discussing this in discord. The idea is that most people are so used to pressing the X button to close and save their changes, that this might become confusing because now it no longer does. The discard button should be more than obvious explaining what it does so that's why there's no confirmation.

**Comment by SnappyComebacks** [comment]
What are your thoughts on the big red button just performing a reset but not closing the dialogue, and the user always has to press the big green button as a general "Accept & Start Campaign"? (Window close X would behave the way you have it behave now).
If that's too hard / or you don't want to do it that's Ok, just let me know and I'll merge what you have here.

**Comment by Raffson** [comment]
> What are your thoughts on the big red button just performing a reset but not closing the dialogue,...

That was up for discussion as well, I could take an extra look but tbh I think it's better off as a separate button. What do you think?

**Comment by SnappyComebacks** [comment]
I lean towards the two button approach (Reset to Default, Accept & Start) as we have tied the color green to the concept of next/proceed.

File: qt_ui/windows/AirWingConfigurationDialog.py [hunk 1]
```diff
@@ -512,13 +513,33 @@ def __init__(self, game: Game, parent) -> None:
...
+    def accept(self) -> None:
         for tab in self.tabs:
             tab.apply()
+        super().accept()
+
+    def reject(self) -> None:
+        result = QMessageBox.information(
+            None,
+            "Discard changes?",
+            "Are you sure you want to discard your changes?",
```
Comment (SnappyComebacks) [inline qt_ui/windows/AirWingConfigurationDialog.py]: This should probably mention that it will start the campaign as well.

Comment (Raffson) [reply qt_ui/windows/AirWingConfigurationDialog.py]: Agreed 👍

**Comment by Raffson** [comment]
> I lean towards the two button approach (Reset to Default, Accept & Start) as we have tied the color green to the concept of next/proceed.

I'll give it a shot...

**Comment by Raffson** [comment]
I'll be taking things a step further, because it'll be annoying every time you don't make any changes and press [X], you'll get that pop-up. I'll be adding a boolean value that'll be set to True if a "change signal" is emitted, that way I know when to show the message. The same thing applies to the revert, saving CPU cycles if the "Discard Changes" button is pressed with no prior changes...

The trickiest part is the actual revert. I have a solution at the moment that works, but I'm not too happy with it because essentially it's a full re-initialization of the tabs...

**Comment by Raffson** [comment]
Took a while, but I finally have something I can live with. It's still pretty much a re-initialization, but at least I'm not getting the double pop-up upon creation of the air wing dialog.

File: qt_ui/windows/AirWingConfigurationDialog.py [hunk 2]
```diff
@@ -502,23 +526,61 @@ def __init__(self, game: Game, parent) -> None:
...
+        layout.addLayout(buttons_layout)
 
-    def reject(self) -> None:
+        self.has_changed = False
+
+    def changed(self) -> None:
+        self.has_changed = True
+
+    def revert(self) -> None:
+        # self.tab_widget.clear()
```
Comment (SnappyComebacks) [inline qt_ui/windows/AirWingConfigurationDialog.py]: Let's remove the commented out code in this method.

Comment (Raffson) [reply qt_ui/windows/AirWingConfigurationDialog.py]: My god, totally forgot about that 😂

**Comment by SnappyComebacks** [comment]
Thanks!

**Comment by Raffson** [comment]
> Thanks!

Gladly done! 😉

**Pull request merged**